compare_mt/reporters.py

```diff
@@ -391,21 +392,39 @@ def print(self):
       sleft, sright = sys_names[left], sys_names[right]
       print(f'--- {report_length} sentences where {sleft}>{sright} at {self.scorer.name()}')
       for bdiff, s1, s2, str1, str2, i in self.scorediff_lists[cnt][:report_length]:
-        print (
-          f"{sleft}-{sright}={fmt(-bdiff)}, {sleft}={fmt(s1)}, {sright}={fmt(s2)}\n"
-          f"Ref:  {' '.join(ref[i])}\n"
-          f"{sleft}: {' '.join(out1[i])}\n"
-          f"{sright}: {' '.join(out2[i])}\n"
-        )
+        if self.src:
+          print (
+            f"{sleft}-{sright}={fmt(-bdiff)}, {sleft}={fmt(s1)}, {sright}={fmt(s2)}\n"
+            f"Src:  {' '.join(self.src[i])}\n"
+            f"Ref:  {' '.join(ref[i])}\n"
+            f"{sleft}: {' '.join(out1[i])}\n"
+            f"{sright}: {' '.join(out2[i])}\n"
+          )
+        else:
+          print (
+            f"{sleft}-{sright}={fmt(-bdiff)}, {sleft}={fmt(s1)}, {sright}={fmt(s2)}\n"
+            f"Ref:  {' '.join(ref[i])}\n"
+            f"{sleft}: {' '.join(out1[i])}\n"
+            f"{sright}: {' '.join(out2[i])}\n"
+          )
 
       print(f'--- {report_length} sentences where {sright}>{sleft} at {self.scorer.name()}')
       for bdiff, s1, s2, str1, str2, i in self.scorediff_lists[cnt][-report_length:]:
-          print (
-          f"{sleft}-{sright}={fmt(-bdiff)}, {sleft}={fmt(s1)}, {sright}={fmt(s2)}\n"
-          f"Ref:  {' '.join(ref[i])}\n"
-          f"{sleft}: {' '.join(out1[i])}\n"
-          f"{sright}: {' '.join(out2[i])}\n"
-        )
+        if self.src:
```

Comment:
I think we should avoid having the duplicated code within the big if statement. We should try to have most of the code not be duplicated, and just the `Src:` part be within an `if`.